### DISCUSSION TITLE
Implementation of Algoburner

### DISCUSSION THREAD

**Discussion by wxivd** [asastats/channel#152] [Ideas]
Token burning affect the total supply of given tokens such as Yieldly, Choice, HDL. 

Accounting for that in the portfolio view would facilitate the purpose of providing accurate information of ASA held in Wallet

What technical difficulties are there for implementation of such information

**Comment by ipaleka** [comment]
We don't have a system for presenting events, that would be the first one, is that worth it?

We can add an additional item of circulating supply to the left, but then that event should be presented only indirectly, by a lower value after it happens.

**Reply by wxivd** [reply]
> We don't have a system for presenting events, that would be the first one, is that worth it?

That's open for discussion. I do not see it as a priority for now.

Initial thoughts were about deducting the amount of Assets sent to the burner account wallet from the total supply. This could be presented as burnt or permanently locked. A downside to this is the amount of manual work and presuming the burner wallet is kept the same for each ASA.

The assumption did not hold true. Each burning is set to new wallet as is shown in Yieldly ([1](https://algoexplorer.io/address/TDMWTZTBYEZTIYM5AWOJ7RMFJ3GG2JXBFHDQX2WY56UTSWIQOSPBE7EYBY) & [2](https://algoexplorer.io/address/ZBWTV6CR4EBLMWHN6ABKPWH35ZQELIF5K2HKC2SABOLSXCMUVZNAHFWFMY))

> We can add an additional item of circulating supply to the left, but then that event should be presented only indirectly, by a lower value after it happens.

How can is be presented indirectly? Something down the line of presenting ASA-list in distinct page similar to [Tinyman ASA List](https://asa-list.tinyman.org/asa-list/) (with handpicked asset details)

**Reply by ipaleka** [reply]
> How can is be presented indirectly?

I mean, just by showing circulating supply value. So if you looked before and then if you look after burning, those two values will differ.

Algorand doesn't have an actual burning mechanism implemented, so projects are using some pseudo-burning mechanism, IIRC when you send an amount to those escrows like Yieldly did in that example, then the circulating supply value decreases.
